File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) {
        if ( (a * b) % 2 == 1) return (int)(Math.pow(a, 2) + Math.pow(b, 2));
        else if (a % 2 == 1 || b % 2 == 1) return 2 * (a + b);
        else if((a * b) % 2 == 0) return Math.abs(a - b);
        else return 0;
    }
}

File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.js
function solution(a, b) {
    return (a * b) % 2 == 1 ? a * a + b * b :
           (a % 2 == 1 || b % 2 == 1) ? 2 * (a + b) :
           (a * b) % 2 == 0 ? Math.abs(a - b) :
           0;
}